src/Compilers/VisualBasic/Portable/Binding/LookupResult.vb

```diff
@@ -691,6 +691,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     CompactSymbols(lost)
                     Debug.Assert(_symList.Count > 0)
 
+                    ' As a special case, we allow conflicting enum members imported from
+                    ' metadata If the have the same value, and take the first
```

Comment:
> It is not clear what makes you think the order of members is deterministic.

I think the order is deterministic because we went to some effort to make it deterministic, and I can see that the code to make that happen still appears in the compilers.